tests/libjlm/opt/test-unroll.cpp

```diff
@@ -270,11 +271,95 @@ test_unknown_boundaries()
+
+	/* Outer loop */
+	auto otheta = jive::theta_node::create(graph.root());
+
+	auto lvo_init = otheta->add_loopvar(init);
+	auto lvo_step = otheta->add_loopvar(step);
+	auto lvo_end = otheta->add_loopvar(end);
+
+	auto add = jive::bitadd_op::create(32, lvo_init->argument(), lvo_step->argument());
+	auto compare = jive::bitult_op::create(32, add, lvo_end->argument());
+	auto match = jive::match(1, {{1, 1}}, 0, 2, compare);
+	otheta->set_predicate(match);
+	lvo_init->result()->divert_to(add);
+
+	/* First inner loop in the original loop */
+	auto inner_theta = jive::theta_node::create(otheta->subregion());
+
+	auto inner_init = jive::create_bitconstant(otheta->subregion(), 32, 0);
+	auto lvi_init = inner_theta->add_loopvar(inner_init);
+	auto lvi_step = inner_theta->add_loopvar(lvo_step->argument());
+	auto lvi_end = inner_theta->add_loopvar(lvo_end->argument());
+
+	auto inner_add = jive::bitadd_op::create(32, lvi_init->argument(), lvi_step->argument());
+	auto inner_compare = jive::bitult_op::create(32, inner_add, lvi_end->argument());
+	auto inner_match = jive::match(1, {{1, 1}}, 0, 2, inner_compare);
+	inner_theta->set_predicate(inner_match);
+	lvi_init->result()->divert_to(inner_add);
+
+	/* Nested inner loop */
+	auto inner_nested_theta = jive::theta_node::create(inner_theta->subregion());
+
+	auto inner_nested_init = jive::create_bitconstant(inner_theta->subregion(), 32, 0);
+	auto lvi_nested_init = inner_nested_theta->add_loopvar(inner_nested_init);
+	auto lvi_nested_step = inner_nested_theta->add_loopvar(lvi_step->argument());
+	auto lvi_nested_end = inner_nested_theta->add_loopvar(lvi_end->argument());
+
+	auto inner_nested_add = jive::bitadd_op::create(32, lvi_nested_init->argument(), lvi_nested_step->argument());
+	auto inner_nested_compare = jive::bitult_op::create(32, inner_nested_add, lvi_nested_end->argument());
+	auto inner_nested_match = jive::match(1, {{1, 1}}, 0, 2, inner_nested_compare);
+	inner_nested_theta->set_predicate(inner_nested_match);
+	lvi_nested_init->result()->divert_to(inner_nested_add);
+	
+	/* Second inner loop in the original loop */
+	auto inner2_theta = jive::theta_node::create(otheta->subregion());
+
+	auto inner2_init = jive::create_bitconstant(otheta->subregion(), 32, 0);
+	auto lvi2_init = inner2_theta->add_loopvar(inner2_init);
+	auto lvi2_step = inner2_theta->add_loopvar(lvo_step->argument());
+	auto lvi2_end = inner2_theta->add_loopvar(lvo_end->argument());
+
+	auto inner2_add = jive::bitadd_op::create(32, lvi2_init->argument(), lvi2_step->argument());
+	auto inner2_compare = jive::bitult_op::create(32, inner2_add, lvi2_end->argument());
+	auto inner2_match = jive::match(1, {{1, 1}}, 0, 2, inner2_compare);
+	inner2_theta->set_predicate(inner2_match);
+	lvi2_init->result()->divert_to(inner2_add);
+
+	
+	jive::view(graph, stdout);
+	jlm::loopunroll loopunroll(4);
+	loopunroll.run(rm, sd);
+	/* FIX ME! Need a way to check that only the inner loop gets unrolled */
```

Comment:
Can we please add these inserts (and remove the FIXMEs)?

Reply:
Not sure how to check for this.
Possibly by counting the number of nodes in the thetas, but it's not really checking for correctness.
Maybe could do check that the number of nodes is greater than the number of nodes of the original body.

Reply:
The checking for (complete) correctness will be messy. I would not advice it. Simply counting the number of nodes is sufficient for now.